mmdet3d/datasets/pipelines/transforms_3d.py

```diff
@@ -867,14 +883,19 @@ def __call__(self, results):
 
         Args:
             input_dict (dict): Result dict from loading pipeline.
-
         Returns:
             dict: Results after sampling, 'points', 'pts_instance_mask' \
                 and 'pts_semantic_mask' keys are updated in the result dict.
         """
+        from mmdet3d.core.points import CameraPoints
         points = results['points']
-        points, choices = self.points_random_sampling(
-            points, self.num_points, return_choices=True)
+        # Points in Camera coord can provide the depth information.
+        # TODO: Need to suport distance-based sampling for other coord system.
+        if self.sample_range:
```

Comment:
Also change to `if self.sample_range is not None` ?

Reply:
Comments resolved. Code refined. Thx.

Reply:
This line is not resolved yet?

Reply:
That's just a reminder for developers that we can only support sampling with distance for camera coord system. If necessary, I will switch it to become a `NOTE` message instead of a `TODO` message.

Reply:
Comments resolved. Thx.